### ISSUE TITLE
Checker found undefined keys false positives with only partial keys

### ISSUE THREAD

**Issue by seedydocloop** [lingualdev/i18n-check#123]
# Summary

When checker checks for undefined keys, it produces false positives and seems to fail to understand my translation keys convention.

## Tech stack

React + react-i18next + i18next with new selector API

### File structure

Here's my translation files structure : several translation namespaces grouped by language folder
```
src/
  locales/
    en/
      file.json
      otherfile.json
      ...
    fr/
      file.json
      otherfile.json
      ...
```

### Translation keys structure

Using flat keys separated by dots.

Example from an existing translation file :
```
{
  "all_docs.upload_file": "Upload file(s)",
  "autocomplete.clear": "Clear",
  "autocomplete.no_options": "No options",
  "autocomplete.toggle_menu": "Toggle menu",
  "back.home": "Back to home",
  "breadcrumb.toggle_menu": "Toggle menu",
  "breadcrumbs_menu.home": "Home",
}
```

## Command

`i18n-check -s en --locales src/locales/ -u src/ -o undefined -f i18next`

### Output

```
Found undefined keys!
┌─────────────────────────────────────────────────────────────────────────────────────┬─────────────────────────────────────────────────────────────────────────────────────┐
│ file                                                                                │ key                                                                                 │
├─────────────────────────────────────────────────────────────────────────────────────┼─────────────────────────────────────────────────────────────────────────────────────┤
│ REDACTED                                            │ registration_failed"]
│ REDACTED                                            │ home"]   
```

### Checking manually

Keys are only partially displayed in the output.
 
**For line 1 :**

There is no `registration_failed` key in my translation files

Culprit seems to be :
```
const { t } = useTranslation(["auth", "validation"]);
// stuff

t(($) => $["registration_view.registration_failed"], { ns: "auth" }),
```

But the translation exists in my `auth.json` file.

**For line 2:**

Same goes for the second line, which seems to match the following :
```
const { t } = useTranslation("common");
// stuff

t(($) => $["back.home"])
```

But the translation exists in my `common.json` file.

**Comment by busypeoples** [comment]
It currently checks for a structure like in the [documentation](https://www.i18next.com/overview/api#typescript-1): `t($ => $.some.key); ` What we can add is  `t(($) => $["some.key"])`.
Could not find all the possible ways that the selector API can be used via the documentation, but we can fix that scenario.

**Comment by seedydocloop** [comment]
> It currently checks for a structure like in the [documentation](https://www.i18next.com/overview/api#typescript-1): `t($ => $.some.key); ` What we can add is `t(($) => $["some.key"])`. Could not find all the possible ways that the selector API can be used via the documentation, but we can fix that scenario.

You are right, this is undocumented. I think this syntax is related to how I generate my translation types, using `i18next-resources-for-ts`, and "optimize" mode.

In case it might help, my generated `Resources` look like this :

```
interface Resources {
common: {
    "all_docs.upload_file": "Upload file(s)";
    "autocomplete.clear": "Clear";
    "autocomplete.no_options": "No options";
    "autocomplete.toggle_menu": "Toggle menu";
    "back.home": "Back to home";
    "breadcrumb.toggle_menu": "Toggle menu";
    "breadcrumbs_menu.home": "Home";
    "collapsible.toggle": "Toggle";
    "confirm.unsaved.changes": "You have unsaved changes";
    "confirm.unsaved.changes.description": "If you leave the current page, they will be lost";
    "confirm.unsaved.changes.leave_discard": "Leave and discard changes";
    "confirm.unsaved.changes.stay": "Stay";
    // ...
   }
}
```

My cli command :
`i18next-resources-for-ts interface --optimize -i ./src/locales/en -o ./src/@typedef/i18next-resources.d.ts`


Thanks for working on this, it is of great help to maintain my translations 🙏

**Comment by busypeoples** [comment]
Should be fixed in [`v0.9.4`](https://github.com/lingualdev/i18n-check/releases/tag/v0.9.4), which supports using the i18next api selector with element access:

```ts
t($ => $["a.b.c"]);
```

**Comment by seedydocloop** [comment]
> Should be fixed in [`v0.9.4`](https://github.com/lingualdev/i18n-check/releases/tag/v0.9.4), which supports using the i18next api selector with element access:
> 
> t($ => $["a.b.c"]);

@busypeoples 

Tested. Because of my code formatter, I sometimes have line-jumps like the following :
```
t($ => $[
  "a.b.c"
  ],
{ ns: "ns" },
)
```

It seems not to be handled by 0.9.4, as the key is listed among unused keys.

Apart from this very peculiar point, the fix works with 90% of my translations, thanks ! 🙏

**Comment by busypeoples** [comment]
We will look into fixing that issue.

**Comment by busypeoples** [comment]
Ok, the formatted element access should also work now and be available in `v0.9.5` @seedydocloop
